src/main/kotlin/agallipo/BoardingCardSorter.kt

```diff
@@ -1,114 +1,114 @@
+    //created by alex :)
+    interface BoardingCardSorter {
 
-interface BoardingCardSorter {
+        //This function divides  all the boarding cards into lists of flight-paths, iterating over all the boarding cards,
+        //it first checks if the boarding card can be joined to any journey, if not, it creates a new element in the
+        //list of flighPaths. Then it looks over if any flight path can be joined
 
-    //This function divides  all the boarding cards into lists of flight-paths, iterating over all the boarding cards,
-    //it first checks if the boarding card can be joined to any journey, if not, it creates a new element in the
-    //list of flighPaths. Then it looks over if any flight path can be joined
-
-    fun sortTickets(boardingCards: MutableList<BoardingCard?>): MutableList<MutableList<BoardingCard?>> {
-        var flightPaths = mutableListOf<MutableList<BoardingCard?>>()
-        for (boardingCard in boardingCards) {
-            if (!matchFlightPath(boardingCard, flightPaths))
-                flightPaths.add(mutableListOf<BoardingCard?>(boardingCard))
-            checkJoinFlightPath(flightPaths)
+        fun sortTickets(boardingCards: MutableList<BoardingCard?>): MutableList<MutableList<BoardingCard?>> {
+            var flightPaths = mutableListOf<MutableList<BoardingCard?>>()
+            for (boardingCard in boardingCards) {
+                if (!matchFlightPath(boardingCard, flightPaths))
+                    flightPaths.add(mutableListOf<BoardingCard?>(boardingCard))
+                checkJoinFlightPath(flightPaths)
+            }
+            /*for (flight in flightPaths){
+                checkJoinFlightPath(flightPaths)
+            }*/
+            return flightPaths
         }
-        /*for (flight in flightPaths){
-            checkJoinFlightPath(flightPaths)
-        }*/
-        return flightPaths
-    }
 
-    //This function iterates over the flight paths and checks,returning a boolean, if the given boarding card as a
-    // parameter can be added at the end or beginning of any journey.
-    fun matchFlightPath(boardingCard: BoardingCard?, flightPaths: MutableList<MutableList<BoardingCard?>>): Boolean {
-        for (path in flightPaths) {
-            if (boardingCard!!.origin == path[0]!!.destination) {
-                path.add(0, boardingCard)
-                return true
-            }
-            if (boardingCard!!.origin == path[path.size - 1]!!.destination) {
-                path.add(boardingCard)
-                return true
+        //This function iterates over the flight paths and checks,returning a boolean, if the given boarding card as a
+        // parameter can be added at the end or beginning of any journey.
+        fun matchFlightPath(boardingCard: BoardingCard?, flightPaths: MutableList<MutableList<BoardingCard?>>): Boolean {
+            for (path in flightPaths) {
+                if (boardingCard!!.origin == path[path.size - 1]!!.destination) {
```

Comment:
Try this: add the annotation of Non nullable in the data class and then check if you need to force it here.